java/compaction/compaction-job-creation/src/main/java/sleeper/compaction/job/creation/CreateCompactionJobs.java

```diff
@@ -121,31 +125,50 @@ public void createJobs(TableProperties table) throws StateStoreException, IOExce
 
     private void createJobsForTable(TableProperties tableProperties, StateStore stateStore) throws StateStoreException, IOException, ObjectFactoryException {
+        Instant startTime = Instant.now();
         TableStatus table = tableProperties.getStatus();
         LOGGER.debug("Creating jobs for table {}", table);
 
+        Instant loadPartitionStartTime = Instant.now();
         List<Partition> allPartitions = stateStore.getAllPartitions();
+        Duration loadPartitionDuration = Duration.between(loadPartitionStartTime, Instant.now());
 
         // NB We retrieve the information about all the active file references and filter
         // that, rather than making separate calls to the state store for reasons
         // of efficiency and to ensure consistency.
+        Instant loadFilesStartTime = Instant.now();
         List<FileReference> fileReferences = stateStore.getFileReferences();
+        Duration loadFilesDuration = Duration.between(loadFilesStartTime, Instant.now());
 
+        Instant compactionStrategyCreationStartTime = Instant.now();
         CompactionStrategy compactionStrategy = objectFactory
                 .getObject(tableProperties.get(COMPACTION_STRATEGY_CLASS), CompactionStrategy.class);
+        Duration objectFactoryCreationDuration = Duration.between(compactionStrategyCreationStartTime, Instant.now());
         LOGGER.debug("Created compaction strategy of class {}", tableProperties.get(COMPACTION_STRATEGY_CLASS));
         CompactionJobFactory jobFactory = new CompactionJobFactory(instanceProperties, tableProperties, jobIdSupplier);
 
+        Instant indexCreationStartTime = Instant.now();
+        CompactionStrategyIndex index = new CompactionStrategyIndex(tableProperties.getStatus(), fileReferences, allPartitions);
+        Duration indexCreationDuration = Duration.between(indexCreationStartTime, Instant.now());
+
+        Instant jobCreationStartTime = Instant.now();
         List<CompactionJob> compactionJobs = compactionStrategy.createCompactionJobs(
-                instanceProperties, tableProperties, jobFactory, fileReferences, allPartitions);
+                instanceProperties, tableProperties, jobFactory, index);
+        Duration jobCreationDuration = Duration.between(jobCreationStartTime, Instant.now());
         LOGGER.info("Used {} to create {} compaction jobs for table {}", compactionStrategy.getClass().getSimpleName(), compactionJobs.size(), table);
 
+        Duration leftoverJobDuration = null;
         if (mode == Mode.FORCE_ALL_FILES_AFTER_STRATEGY) {
+            Instant leftoverJobCreationStartTime = Instant.now();
             createJobsFromLeftoverFiles(tableProperties, jobFactory, fileReferences, allPartitions, compactionJobs);
+            leftoverJobDuration = Duration.between(leftoverJobCreationStartTime, Instant.now());
         }
         int creationLimit = instanceProperties.getInt(CompactionProperty.COMPACTION_JOB_CREATION_LIMIT);
+        Duration limitJobsDuration = null;
```

Comment:
For the Durations that are nullable, it might be worth defaulting the time after to the same as the time before, so we don't need nulls?